### PR TITLE
Board Game Pack!

### PR DESCRIPTION
**Pull request by Laugic**
Board game pack!

### CODE REVIEW

File: src/main/java/thePackmaster/actions/boardgamepack/DelayedDiceLevelUpAction.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+import thePackmaster.powers.boardgamepack.DicePower;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static thePackmaster.util.Wiz.adp;
+
+public class DelayedDiceLevelUpAction extends AbstractGameAction {
+
+    public DelayedDiceLevelUpAction() {
```
Comment (erasels) [inline src/main/java/thePackmaster/actions/boardgamepack/DelayedDiceLevelUpAction.java:15]: Could be an anonymous action, they help me keep the project to a manageable size

File: src/main/java/thePackmaster/actions/boardgamepack/RollDiceAction.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+    public void update() {
+        //ADD DICE ROLL VISUALS HERE
+//        int diceTotal = 0;
+        for(int numDice = 0; numDice < dice.size(); numDice++)
+        {
+            int curRoll = 0;
+            for(int roll = 0; roll <= advantage; roll++)
+            {
+                int newRoll = AbstractDungeon.miscRng.random(dice.get(numDice) - 1) + 1;
```
Comment (erasels) [inline src/main/java/thePackmaster/actions/boardgamepack/RollDiceAction.java:36]: Wouldn't cardRandomRng fit better here? It's used for basically all impactful mid-combat randomization

File: src/main/java/thePackmaster/actions/boardgamepack/RollDiceAction.java [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+//            AbstractDungeon.actionManager.addToBottom(new ApplyPowerAction(
+//                    AbstractDungeon.player, AbstractDungeon.player, new DicePower(AbstractDungeon.player, curRoll, dice.get(numDice)), curRoll));
+//            diceTotal += curRoll;
+        }
+
+        //int rand = new Random().nextInt(4);
+        //addToTop(new SFXAction(modID + ":dice1"));
+
+//        AbstractDungeon.actionManager.addToBottom(new ApplyPowerAction(
+//                AbstractDungeon.player, AbstractDungeon.player, new DicePower(AbstractDungeon.player, diceTotal, diceSize), diceTotal));
```
Comment (erasels) [inline src/main/java/thePackmaster/actions/boardgamepack/RollDiceAction.java:49]: Is this action done? I'm not sure if I understand it

File: src/main/java/thePackmaster/cards/boardgamepack/FlickDice.java [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+    public FlickDice() {
+        super(ID, 1, CardType.ATTACK, CardRarity.COMMON, CardTarget.ENEMY);
+        damage = baseDamage = 9;
+    }
+
+    public void use(AbstractPlayer p, AbstractMonster m) {
+        Wiz.doDmg(m, damage);
+        //addToBot(new RollDiceAction(upgraded?UP_DICE:DICE));
+        int dice = upgraded?8:6;
```
Comment (erasels) [inline src/main/java/thePackmaster/cards/boardgamepack/FlickDice.java:29]: I take it, magic number wouldn't work here?

File: src/main/java/thePackmaster/cards/boardgamepack/FlickDice.java [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+        addToBot(new ApplyPowerAction(p, p, new DicePower(p, dice), dice));
+    }
+
+    public void upp() {
+        upgradeDamage(3);
+        uDesc();
```
Comment (erasels) [inline src/main/java/thePackmaster/cards/boardgamepack/FlickDice.java:35]: upp already handles description upgrading

File: src/main/resources/anniv5Resources/localization/eng/boardgamepack/Powerstrings.json [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+    "DESCRIPTIONS": [
+      "Skip your next turn.",
+      "Skip your next #b",
+      " turns."
+    ]
+  },
+  "${ModID}:OneTimeAdvantagePower": {
+    "NAME": "Advantage",
+    "DESCRIPTIONS": [
+      "Roll the next Dice an additional time and take the highest value.",
```
Comment (erasels) [inline src/main/resources/anniv5Resources/localization/eng/boardgamepack/Powerstrings.json:20]:
```suggestion
      "Roll the next #yDice an additional time and take the highest value.",
```
Maybe?

**Review by erasels — CHANGES_REQUESTED** [review]

File: src/main/java/thePackmaster/actions/boardgamepack/RollDiceAction.java (hunk 3 above)
Comment (Laugic) [reply src/main/java/thePackmaster/actions/boardgamepack/RollDiceAction.java:49]: I should have removed that sorry, the functionality was moved to the dice power

File: src/main/java/thePackmaster/cards/boardgamepack/AbstractBoardCard.java [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+
+    public boolean reroll = false, isChance = false;
+
+    public AbstractBoardCard(String cardID, int cost, CardType type, CardRarity rarity, CardTarget target) {
+        super(cardID, cost, type, rarity, target);
+        setBackgroundTexture(
+//                "anniv5Resources/images/512/boardgame/board-back-" + type.name().toLowerCase() + ".png",
+//                "anniv5Resources/images/1024/boardgame/board-back-" + type.name().toLowerCase() + ".png"
+                "anniv5Resources/images/512/boardgame/board-back-3.png",
+                "anniv5Resources/images/1024/boardgame/board-back-3.png"
```
[the remark this answers is not in the document]
Comment (Laugic) [reply src/main/java/thePackmaster/cards/boardgamepack/AbstractBoardCard.java:15]: I was doing lots of experimenting to try and see which ones looked good. I should clean those up to reduce the size

File: src/main/java/thePackmaster/cards/boardgamepack/FlickDice.java (hunk 4 above)
Comment (Laugic) [reply src/main/java/thePackmaster/cards/boardgamepack/FlickDice.java:29]: I didn't want the number to go up from other card effects that modified the magic number without it showing up on the card (and also the dice are only certain sizes)

File: src/main/java/thePackmaster/cards/boardgamepack/FlickDice.java (hunk 5 above)
Comment (Laugic) [reply src/main/java/thePackmaster/cards/boardgamepack/FlickDice.java:35]: I tried it without uDesc and it didn't seem to work- maybe it was a fluke. I'll try removing it

File: src/main/java/thePackmaster/actions/boardgamepack/DelayedDiceLevelUpAction.java (hunk 1 above)
Comment (Laugic) [reply src/main/java/thePackmaster/actions/boardgamepack/DelayedDiceLevelUpAction.java:15]: I'll give that a try

File: src/main/resources/anniv5Resources/localization/eng/boardgamepack/Powerstrings.json (hunk 6 above)
Comment (Laugic) [reply src/main/resources/anniv5Resources/localization/eng/boardgamepack/Powerstrings.json:20]: maybe

**Pull request merged**